raytracing/__main__.py

```diff
@@ -9,28 +9,50 @@
 from . import olympus
 from . import utils
 
+import os 
+import sys
 import argparse
+import re
+
+pythonExec = sys.executable        
+topDir = os.path.dirname(os.path.realpath(__file__))
+examplesDir = "{0}/../examples".format(topDir)
+exampleScriptNames = [f for f in os.listdir(examplesDir) if re.match(r'.*.py$', f)]
+exampleScriptNames.sort()
+exampleScripts = {name:"{0}/{1}".format(examplesDir, name) for name in exampleScriptNames}
 
 ap = argparse.ArgumentParser(prog='python -m raytracing')
 ap.add_argument("-e", "--examples", required=False, default='all',
                 help="Specific example numbers, separated by a comma")
 ap.add_argument("-c", "--classes", required=False, action='store_const',
                 const=True, help="Print the class hierarchy in graphviz format")
+ap.add_argument("-l", "--list", required=False, action='store_const',
+                const=True, help="List all the accessible examples")
 
 args = vars(ap.parse_args())
 examples = args['examples']
 printClasses = args['classes']
+listExamples = args['list']
 
 if printClasses:
     printClassHierarchy(Rays)
     printClassHierarchy(Matrix)
     exit(0)
-
+elif listExamples:
```

Comment:
Why `elif` ? there doesn't seem to be a constraint that prohibits the user of doing `-r -c`. If that is the case, we should see a `MutuallyExclusiveGroup` as the argparse argument parser class

Reply:
It makes more sense to have only one. Not sure how to make a group.

Reply:
I just modified it and it gets weird because -e defaults to all so we don't see the list, we just see a window popping up.  I am leaving the elif